Polygon_mesh_processing/include/CGAL/Polyhedral_envelope.h

```diff
@@ -0,0 +1,1910 @@
+  };
+
+  template <class Kernel>
+  struct Point_map
+  {
+    typedef boost::readable_property_map_tag category;
+    typedef std::size_t key_type;
+    typedef typename Kernel::Point_3 value_type;
+    typedef value_type reference;
+
+    const std::vector<Iso_cuboid_3>* boxes_ptr;
+
+    Point_map() : boxes_ptr(nullptr) {}
+    Point_map(const std::vector<Iso_cuboid_3>& boxes) : boxes_ptr(&boxes) {}
+
+    friend value_type get(const Point_map& m, key_type k)
+    {
+      return ((*m.boxes_ptr)[k].min)();
+    }
+  };
+
+  typedef AABB_primitive<std::size_t, Datum_map<GeomTraits>, Point_map<GeomTraits>, Tag_true /*UseSharedData*/, Tag_false /*CacheDatum*/> Primitive;
+  typedef AABB_traits<GeomTraits, Primitive> Tree_traits;
+  typedef AABB_tree<Tree_traits> Tree;
+
+
+  Tree tree;
+
+  eOriented_side_3 oriented_side;
+
+public:
+
+#ifndef DOXYGEN_RUNNING
+  Polyhedral_envelope(const std::vector<Point_3>& env_vertices,
+                      std::vector<Vector3i> env_faces,
+                      double epsilon)
+    : env_vertices(env_vertices), env_faces(env_faces)
+  {
+    init(epsilon);
+  }
+#endif
+
+   /**
+   * Constructor with a triangulated surface mesh.
+    * @tparam TriangleMesh a model of `HalfedgeListGraph`
+    * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+    *
+    * @param tmesh a triangle mesh
+    * @param epsilon the distance of the Minkowski sum hull
+    * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+    *
+    * \cgalNamedParamsBegin
+    *   \cgalParamNBegin{vertex_point_map}
+    *     \cgalParamDescription{a property map associating points to the vertices of `pmesh`}
+    *     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+    *                    as key type and `%Point_3` as value type}
+    *     \cgalParamDefault{`boost::get(CGAL::vertex_point, pmesh)`}
+    *     \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+    *                     must be available in `PolygonMesh`.}
+    *   \cgalParamNEnd
+    *
+    *   \cgalParamNBegin{vertex_index_map}
+    *     \cgalParamDescription{a property map associating to each vertex of `tmesh` a unique index between `0` and `num_vertices(tmesh) - 1`}
+    *     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<TriangleMesh>::%vertex_descriptor`
+    *                    as key type and `std::size_t` as value type}
+    *     \cgalParamDefault{an automatically indexed internal map}
+    *   \cgalParamNEnd
+    *
+    * \cgalNamedParamsEnd
+    *
+    * \note The triangle mesh gets copied internally, that is it can be modifed after having passed as argument,
+    *       while the queries are performed
+    */
+
+  template <typename TriangleMesh, typename NamedParameters>
+  Polyhedral_envelope(const TriangleMesh& tmesh,
```

Comment:
You should add the range of faces to consider as input (and/or add another overload)

Reply:
Done